### PR TITLE
Add support for the Lookup API

### PR DESCRIPTION
**Pull request by samwierema**

### CODE REVIEW

File: MessageBird/Client.cs [hunk 1]
```diff
@@ -104,5 +104,36 @@ public Objects.Balance Balance()
 
             return result.Object as Objects.Balance;
         }
+
+        public Objects.Lookup DoLookup(long phonenumber, LookupOptionalArguments optionalArguments = null)
+        {
+            ParameterValidator.IsNotNullOrWhiteSpace(phonenumber.ToString(), "phonenumber");
```
Comment (rvermeulen) [inline MessageBird/Client.cs]: This check is superfluous

File: MessageBird/Client.cs [hunk 2]
```diff
@@ -104,5 +104,36 @@ public Objects.Balance Balance()
...
+
+            var lookup = new Resources.Lookup(new Objects.Lookup(phonenumber, optionalArguments));
+            var result = restClient.Retrieve(lookup);
+
+            return result.Object as Objects.Lookup;
+        }
+
+        public Objects.LookupHlr RequestLookupHlr(long phonenumber, string reference, LookupHlrOptionalArguments optionalArguments = null)
+        {
+            ParameterValidator.IsNotNullOrWhiteSpace(phonenumber.ToString(), "phonenumber");
```
Comment (rvermeulen) [inline MessageBird/Client.cs]: This check is superfluous as well

File: MessageBird/Client.cs [hunk 3]
```diff
@@ -104,5 +104,36 @@ public Objects.Balance Balance()
...
+
+            var lookupHlr = new Resources.LookupHlr(new Objects.LookupHlr(phonenumber, reference, optionalArguments));
+            var result = restClient.Create(lookupHlr);
+
+            return result.Object as Objects.LookupHlr;
+        }
+
+        public Objects.LookupHlr ViewLookupHlr(long phonenumber, LookupHlrOptionalArguments optionalArguments = null)
+        {
+            ParameterValidator.IsNotNullOrWhiteSpace(phonenumber.ToString(), "phonenumber");
```
Comment (rvermeulen) [inline MessageBird/Client.cs]: This check is superfluous

**Comment by samwierema** [comment]
Thanks for the comments @rvermeulen! Removed the superfluous checks.

@prep also changed the `DoLookup` method to `ViewLookup`.

**Pull request merged**